### PR TITLE
much better armor

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Massively imrpoves armor to no longer be worthless as a all or nothing stat
Massively nerfs armor to no longer be an all or nothing stat

### CODE REVIEW

**Comment by ThePainkiller** [comment]
This won't make practice bullets actually hurt, would it...?

**Comment by r4d6** [comment]
> This won't make practice bullets actually hurt, would it...?

Also won't this allow stuff to kill Bluespace Techs?
Because we still need stuff with 100 armor in something to be completely immune to that stuff, say the BIO armor or the RAD armor...

**Comment by lolman360** [comment]
armor should be a %age reduction tbh instead of whatever the fuck we have now

**Comment by r4d6** [comment]
I did some math, and here's what actually happen : 

Imagine that you have a gun that does 50 damage, and hit against 50 armor, with 10 AP.
First, it calculate the guaranteed_damage_red by multiplying the armor by 0.1, so 50 * 0.1 = 5
Then, it calculate the effectiveness of the armor, 50 armor - 10 AP = 40 armor_effectiveness.
Afterward, it start to calculate the effective damage, first by removing guaranteed_damage_red from the first damage, so it become 50 - 5 = 45 effective damage.
Then, once it goes over stuff like HALLOSS and AGONY damage and calculate the over-penetration, it calculate the actual damage it receive using this formula : 
var/actual_damage = round ( ( effective_damage * ( 100 - armor_effectiveness ) ) / 100 )
If we plug in the numbers, we get var/actual_damage = round ( ( 45 * ( 100 - 40 ) ) / 100 ), which give us the actual damage received of 27.
Then it apply the 27 damage to the player.

File: code/modules/mob/living/living_defense.dm [hunk 1]
```diff
@@ -13,9 +14,13 @@
 	//GDR - guaranteed damage reduction. It's a value that deducted from damage before all calculations
 	var/armor = getarmor(def_zone, attack_flag)
 	var/guaranteed_damage_red = armor * ARMOR_GDR_COEFFICIENT
+	var/guaranteed_damage_blue = damage * ANTI_UNKILLABLE_MOD
 	var/armor_effectiveness = max(0, ( armor - armour_pen ) )
 	var/armor_over_penitration = armour_pen - armor // This basiclly lets us over penitrate to deal extra damage. 20 AP - 10 Armor would give 10, well the reverse would be -10
-	var/effective_damage = damage - guaranteed_damage_red
+	//First we gather are data on damage, this number is raw, so we work with it as is, then we add are armor red with raw armor.
+	//We also have a flat 5% (Lets say a bullet deals 100 damage and they have 200 armor, it still deals 5 damage) we want to have always, this is to avoid people tanking up on 120 armor values and just killdozering the whole game with out ever stopping for short breaks
```
Comment (r4d6) [inline code/modules/mob/living/living_defense.dm:21]: But armor at 100 is _**supposed**_ to block everything.

**Comment by r4d6** [comment]
I just tested it on the live server, and posted the result on discords : 
Calling the proc `damage_through_armor()` directly while playing a Bluespace Tech, which has 100 armor, gave me the following results : 
- 50 damage with 0 AP = 0 damage.
- 50 damage with 50 AP = 20 damage.
- 50 damage with 100 AP = 40 damage.

The GRD was 100 * 0.1 = 10.

So the 50 with 50 AP was reduced to 40, then cut in half.
50 with 100 AP was reduced to 40, then went through
All of which is in line with what the proc is doing.

So if there's something messing with the damage values, it isn't there.

File: code/modules/mob/living/living_defense.dm (hunk 1 above)
Comment (r4d6) [reply code/modules/mob/living/living_defense.dm:21]: Also players can't get armor over 100.

**Comment by Trilbyspaceclone** [comment]
R4 debooonked me and wishes it closed, thus it shall

**Pull request closed without merging**